### ISSUE TITLE
Fail to run l3build for the second time on Windows

### ISSUE THREAD

**Issue by lvjr** [latex3/l3build#185]
I'm trying `l3build` on Windows, but it always produces errors in the second run.

For example, I first download [tabu source code](https://github.com/tabu-issues-for-future-maintainer/tabu), then I can run **`l3build check`** to get the results:
```` 
Check failed with difference files
  - ./build/test/github-0003.pdftex.fc
  - ./build/test/github-0004.pdftex.fc
  - ./build/test/github-0005.pdftex.fc
  - ./build/test/github-0007.pdftex.fc
  - ./build/test/github-0011.pdftex.fc
  - ./build/test/github-0013.pdftex.fc
  - ./build/test/sx87144.pdftex.fc`
 ````
But when I run **`l3build check`** for the second time, I get the follow error:
````
...exmfs/install/scripts/l3build/l3build-file-functions.lua:271: cannot open build/test/sx471000b.lvt: No such file or directory
````
If I try to run **`l3build clean`**, I will get the following error:
````
...exmfs/install/scripts/l3build/l3build-file-functions.lua:271: cannot open build/test/sx430973.pdftex.log: No such file or directory
````
I‘m using MiKTeX 21.2, and l3build 2020-06-04.

**Comment by u-fischer** [comment]
It seems to be a miktex specific problem. With texlive I have no problems, but miktex can't delete files and errors if it should write a file that already exists.

**Comment by josephwright** [comment]
OK, @u-fischer  has spotted the issue: MiKTeX omits `.` and `..` from the directory listings. We probably can deal with that by overloading, with a single test to see if they are returned.

**Comment by u-fischer** [comment]
I opened an issue for miktex https://github.com/MiKTeX/miktex/issues/762

**Comment by LogCreative** [comment]
The problem has not been solved yet. I have installed MiKTeX 21.8 (should be fixed since 21.6 according to MiKTeX/miktex#762).
Download [examples](https://github.com/latex3/l3build/tree/main/examples) and try to build on Windows, still get an error.
```
...ograms/MiKTeX/scripts/l3build/l3build-file-functions.lua:278: cannot open build/test/ascii.tcx: No such file or directory
```
